src/main/kotlin/org/rust/ide/annotator/RsCargoCheckAnnotator.kt

```diff
@@ -82,9 +82,12 @@ class RsCargoCheckAnnotator : ExternalAnnotator<CargoCheckAnnotationInfo, CargoC
             val primarySpan = message.spans
                 .filter { it.is_primary }
                 .filter { it.isValid() }
-                .filter { filePath.endsWith(PathUtil.getCanonicalPath(it.file_name)) }
```

Comment:
@vasily-kirichenko is there any particular reason why we need `getCanonicalPath` here? It is a somewhat expensive operation, because it resolves symlinks, and thus required a syscall per span.